test/cookbooks/selinux_policy_test/recipes/fcontext_filetype.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Cookbook Name:: selinux_policy_test
+# Recipe:: fcontext_filetype
+#
+# Copyright 2016, BackSlasher
+#
+# GPLv2
+
+# Define a single fcontext
+ruby_block 'fail-mismatch' do
+  action :nothing
+  block do
+    raise 'Fail block was invoked'
+  end
+end
+
+basedir = '/var/www'
+symlink = '/var/www/tester_symlink'
+context_dir = 'httpd_sys_content_t'
+context_dir2 = 'httpd_tmp_t'
+context_symlink = 'httpd_sys_rw_content_t'
+
+selinux_policy_fcontext basedir do
+  secontext context_dir
+  filetype 'd'
+end
+
+directory dir_name do
+  recursive true
+end
+
+# Should adapt the symlink itself
+selinux_policy_fcontext symlink do
+  secontext context_symlink
+  filetype 'l'
+end
+
+link symlink do
+  to basedir
+end
+
+# Fail if dir isn't actually set
+execute 'true' do
+  not_if "stat -c %C #{basedir} | grep #{context_dir}"
+  notifies :run, 'ruby_block[fail-mismatch]', :immediate
+end
+
+# Should not run again
+selinux_policy_fcontext 'nomod' do
+  file_spec basedir
+  action :modify
+  secontext context
+  notifies :run, 'ruby_block[fail-mismatch]', :immediate
+end
+
+selinux_policy_fcontext 'modme' do
+  file_spec basedir
+  action :modify
+  secontext context_dir2
+end
+
+# Fail if dir hasn't modified context
+execute 'true' do
+  not_if "stat -c %C #{basedir} | grep #{context_dir2}"
+  notifies :run, 'ruby_block[fail-mismatch]', :immediate
+end
+
+# Fail if symlink has changed context
+execute 'true' do
+  not_if "stat -c %C #{symlink} | grep #{context_symlink}"
+  notifies :run, 'ruby_block[fail-mismatch]', :immediate
+end
+
+# Should not run again
+selinux_policy_fcontext 'symlink-nomod' do
+  file_spec symlink
+  action :modify
+  secontext context_symlink
+  filetype 'l'
+  notifies :run, 'ruby_block[fail-mismatch]', :immediate
+end
+
+selinux_policy_fcontext 'deleteme' do
+  file_spec dir_name
+  filetype 'd'
+  action :delete
+end
+
+selinux_policy_fcontext 'deleteme_symlink' do
+  file_spec symlink
+  filetype 'l'
```

Comment:
I've added a test that should indicate what should happen. However, some "tests" still fail, because the relabel isn't done right away, so restorecon isn't executed. I have no idea based on the current test setup whether I should add relabal actions manually so it's triggered immediately, or if something's wrong with the new test recipe. Can you give feedback on this please?